resources/views/backend/books/create.blade.php

```diff
@@ -34,6 +34,16 @@
               <div class="form-group">
                 <label for="name">{{ __('book.category') }}</label>
                 <select name="category_id" id="category">
+                @foreach($categories as $category)
+                  @if($category->id == App\Model\Book::DEFAULT_CAGEGORY)
+                    @continue;
+                  @endif
+                  @if(old('category_id') == $category->id)
+                    <option selected value="{{ $category->id }}">{{ $category->name }}</option>
```

Comment:
u should put if into option 
ex: @foreach($categories as $category => $value)
                  @if($category->id == App\Model\Book::DEFAULT_CAGEGORY)
                     @continue;
                  @endif
                  <option selected value="{{ $category }}" {{$category == old('category_id')? 'selected' : ''}}>{{ $value }}</option>